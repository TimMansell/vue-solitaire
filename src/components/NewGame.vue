<template>
  <Button @click="newGame" data-test="new-game-btn">
    New Game
  </Button>
</template>

<script>
import { mapActions } from 'vuex';
import Button from './Button.vue';

export default {
  name: 'NewGame',
  components: {
    Button,
  },
  methods: {
    ...mapActions(['restartGame', 'initGame']),
    newGame() {
      this.restartGame();
      this.initGame();
    },
  },
};
</script>
