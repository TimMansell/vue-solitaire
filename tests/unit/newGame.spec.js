import { shallowMount } from '@vue/test-utils';
import NewGame from '@/components/NewGame.vue';

describe('NewGame.vue', () => {
  it('matches snapshot', () => {
    const wrapper = shallowMount(NewGame);

    expect(wrapper).toMatchSnapshot();
  });

  it('calls restartGame when clicked', () => {
    const wrapper = shallowMount(NewGame);
    const mockFunction = jest.fn();

    wrapper.setMethods({ restartGame: mockFunction });

    wrapper.find('[data-test="new-game"]').trigger('click');

    expect(mockFunction).toHaveBeenCalled();
  });
});
