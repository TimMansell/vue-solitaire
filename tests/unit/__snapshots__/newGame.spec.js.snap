// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewGame.vue matches snapshot 1`] = `
<button data-test="new-game-btn" class="new-game__btn">
  New Game
</button>
`;
